lib/phoenix_live_view/html_formatter.ex

```diff
@@ -554,6 +556,22 @@ defmodule Phoenix.LiveView.HTMLFormatter do
 
   defp may_set_preserve(buffer, _text), do: buffer
 
+  defp may_set_preserve_on_text([{:text, text, meta}], tag_name)
+       when tag_name in @inline_elements and tag_name != "button" do
+    mode =
+      if whitespace_around?(text) and meta.newlines == 0 do
```

Comment:
I think we could check `meta.newlines == 0` first to save some `:binary` calls.

```suggestion
      if meta.newlines == 0 and whitespace_around?(text) do
```